components/org.wso2.carbon.identity.sso.saml/src/test/java/org/wso2/carbon/identity/sso/saml/util/SAMLSSOUtilTest.java

```diff
@@ -18,18 +18,173 @@
+
+        prepareForGetIssuer();
+        Issuer issuer = SAMLSSOUtil.getIssuerFromTenantDomain(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME);
+        assertEquals(issuer.getValue(), TestConstants.LOACALHOST_DOMAIN, "Issuer for Super tenant domain.");
+    }
+
+    @Test
+    public void testUnmarshall() throws Exception {
+
+        XMLObject xmlObject = SAMLSSOUtil.unmarshall(TestConstants.DECODED_POST_AUTHN_REQUEST);
+        assertEquals(xmlObject.getDOM().getAttributeNode("Destination").getValue(),
+                "https://localhost:9443/samlsso",
+                "Destination node value of unmarshalled Post Authentication Request is as not expected.");
+        assertEquals(xmlObject.getDOM().getAttributeNode("Reason").getValue(), "Single Logout",
+                "Reason node value of unmarshalled Post Authentication Request is as not expected.");
+    }
+
+    @Test
+    public void testMarshall() throws Exception {
+
+        assertEquals(SAMLSSOUtil.marshall(SAMLSSOUtil.unmarshall(TestConstants.DECODED_POST_AUTHN_REQUEST)),
+                TestConstants.DECODED_POST_AUTHN_REQUEST,
+                "Marshalled Post Authentication Request is not as expected.");
+    }
+
+    @Test
+    public void testEncode() throws Exception {
+
+        String encodedXml = SAMLSSOUtil.encode(TestConstants.AUTHN_FAILED_SAML_RESPONSE);
+        assertFalse(encodedXml.isEmpty(), "Encoded xml should not be empty.");
+        assertFalse(encodedXml.contains(" "), "Encoded xml should not contain spaces.");
+    }
+
+    @Test(expectedExceptions = IdentityException.class)
+    public void testDecode() throws Exception {
+
+        assertEquals(SAMLSSOUtil.decode(TestConstants.ENCODED_REDIRECT_AUTHN_REQUEST),
+                TestConstants.DECODED_REDIRECT_AUTHN_REQUEST,
+                "Decoded value of encoded Redirect Authentication Request is not as expected.");
+        SAMLSSOUtil.decode(TestConstants.GENERAL_STRING);
+    }
+
+    @Test
+    public void testDecodeForPost() throws Exception {
+
+        assertEquals(SAMLSSOUtil.decodeForPost(TestConstants.ENCODED_POST_AUTHN_REQUEST),
+                TestConstants.DECODED_POST_AUTHN_REQUEST,
+                "Decoded value of encoded Post Authentication Request is not as expected.");
+    }
+
+    @DataProvider(name = "testCompressResponse")
+    public static Object[][] compressStrings() {
+
+        return new Object[][] {
+                {TestConstants.AUTHN_FAILED_SAML_RESPONSE,
+                        "Compressed Authentication Failed SAML response should not be empty."},
+                {TestConstants.GENERAL_STRING,
+                        "Compressed general string should not be empty."}
+        };
+    }
+
+    @Test(dataProvider = "testCompressResponse")
+    public void testCompressResponse(String stringToCompress, String message) throws Exception {
+
+        assertFalse(SAMLSSOUtil.compressResponse(stringToCompress).isEmpty(), message);
```

Comment:
Shall we add a description here as well ?

Reply:
here there is a description. 
See 'message'
message is provided via DataProvider.